app/api/resources/user.py

```diff
@@ -58,11 +58,22 @@ def get(cls):
         available_to_mentor. The current user's details are not returned.
         """
 
-        page = request.args.get("page", default=UserDAO.DEFAULT_PAGE, type=int)
-        per_page = request.args.get("per_page", default=UserDAO.DEFAULT_USERS_PER_PAGE, type=int)
+        need_mentoring = request.args.get("need_mentoring")
+        available_to_mentor = request.args.get("available_to_mentor")
+        if need_mentoring and need_mentoring.lower() == "true":
```

Comment:
Are you considering that these 2 fields have to have a value to consider them for filtering? You can verify in this condition only `need_mentoring` and check `available_to_mentor` in the condition following this one. It should be fine to only filter by one of these attributes @aditya2499 
Let me know if you have any doubt :)

Reply:
There is no compulsion to pass the values to any of these attributes in the request. If the values of there attributes are not passed, `request.args.get` will give the null. API will only filter result based on the attribute when only `true` value is passed, in other cases( like `false` value, no value is passed, a random value), it will return the result without considering the attribute (like it is working presently).

@isabelcosta  Let me know if I am not clear, I will add the screenshots of the dry-run and explain possible cases.